dao/application_authentication_dao.go

```diff
@@ -11,6 +11,46 @@ type ApplicationAuthenticationDaoImpl struct {
 	TenantID *int64
 }
 
+func (a *ApplicationAuthenticationDaoImpl) ApplicationAuthenticationsByApplications(applications []m.Application) ([]m.ApplicationAuthentication, error) {
+	var applicationAuthentications []m.ApplicationAuthentication
+
+	applicationIDs := make([]int64, 0)
+	for _, value := range applications {
+		applicationIDs = append(applicationIDs, value.ID)
+	}
+
+	result := DB.Preload("Tenant").Where("application_id IN ?", applicationIDs).Find(&applicationAuthentications)
+	if result.Error != nil {
+		return nil, result.Error
+	}
```

Comment:
You can append `.Error` after the last call so that it directly returns the error. That way you can simply do a `if err != nil { return nil, err`.